AmplifyPlugins/API/AWSAPICategoryPlugin/Operation/AWSHTTPURLResponse.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// `AWSHTTPURLResponse` contains the response body and metadata associated with the response to an HTTP request.
+///
+/// When using `AWSAPIPlugin`, you can optionally type cast `HTTPURLResponse` to an instances of `AWSHTTPURLResponse`.
+/// The response body can be accessed from the `body: Data?` property. For example, when the `APIError` is an
+/// `.httpStatusError(StatusCode, HTTPURLResponse)`, then access the response body by type casting the response to
+/// an `AWSHTTPURLResponse` and retrieve the `body` field.
+/// ```swift
+/// if case let .httpStatusError(statusCode, response) = error, let awsResponse = response as? AWSHTTPURLResponse {
+///     if let responseBody = awsResponse.body {
+///         print("Response contains a response body")
+///     }
+/// }
+/// ```
+/// **Note**: The class inheritance to `HTTPURLResponse` is to provide above mechanism, and actual
+/// implementation acts as a facade that stores an instance of `HTTPURLResponse` that delegates overidden methods to
+/// this stored property.
+public class AWSHTTPURLResponse: HTTPURLResponse {
+
+    /// The response body of the response
+    public let body: Data?
+
+    private let response: HTTPURLResponse
+
+    init?(response: HTTPURLResponse, body: Data?) {
+        self.body = body
+        self.response = response
+
+        // Call the super class initializer with dummy values to satisfy the requirement of the inheritance.
+        super.init(url: URL(string: "dummyURL")!,
+                   statusCode: 0,
+                   httpVersion: nil,
+                   headerFields: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented and is not expected to be called.")
```

Comment:
Why can't we delegate to super instead of crashing?

```suggestion
        super.init(coder: coder)
```

Reply:
yes we can, this change appears to be the safer option for sure